### ISSUE TITLE
Refactor ZairaChem modules to Ersilia Model Hub components

### ISSUE THREAD

**Issue by miquelduranfrigola** [ersilia-os/zaira-chem#30]
## Motivation

At the moment, [ZairaChem](https://github.com/ersilia-os/zaira-chem) has a lot of dependencies, as apparent from the `install.sh` file. Importantly, several Conda environments are created, which makes it difficult/impossible to maintain. To make ZairaChem more sustainable, we need to migrate most of its code to Ersilia Model Hub artefacts.

## Types of ZairaChem elements

There are 2 types of modules we want to migrate:
- **Static**: These are the easy ones. For example, ZairaChem uses MELLODDY-Tuner to convert a list of SMILES into a normalized form, with some extra columns. In principle, we could create a new Ersilia model (called, for example, `melloddy-tuner`) where this is done in an isolated container/environment. This would prevent us from having to install MELLODDY and its dependencies.
- **Trainable**: ZairaChem uses AutoML frameworks such as FLAML, AutoGluon, Keras Tuner and others. These frameworks are used to automatically train models based on descriptors. Ideally, we want to migrate these trainers into Ersilia Model Hub artefacts. The main challenge 😟 is that, at the moment, Ersilia does not accept `fit` instructions. Therefore, we would need to figure this out first. At a high level, we'd like to have fitting capabilities at training time, acompanied with some persistency of AI models in order to use them at prediction time.

## Roadmap

We should start with **static** migration, while we figure out the approach for **trainable** models. I suggest the following order (subject to change):

- [ ] Ersilia Compound Embeddings (eos2gw4)
- [ ] MELLODDY-Tuner
- [ ] Modify Ersilia codebase to enable `fit` commands. Probably, we should work on this in a separate issue.
- [ ] Create a fittable FLAML Ersilia model.
- [ ] Create a fittable Keras Tuner Ersilia model.
- [ ] Create a fittable AutoGluon Ersilia model.
- [ ] Create a fittable MolMap Ersilia model.
- [ ] Replace existing code in ZairaChem by `ErsiliaModel` Python API calls.

**Comment by GemmaTuron** [comment]
We will need to complete the refactoring before fully incorporating #31 @miquelduranfrigola

**Comment by miquelduranfrigola** [comment]
Yes, all clear. Tagging @DhanshreeA since we will have to further improve input output adapters, and most important, figure out a way to make Ersilia models trainable and fine-tunable.

**Comment by GemmaTuron** [comment]
Update on this task. We have decided to Dockerize each step of Zairachem and then simply orchestrate the containers. The steps are clearly defined by ZairaChem's pipeline. We will be working on this repository: https://github.com/zairachem-docker

Along the process we will work in improving some dependency issues such as removing altogether Melloddy. I will close this issue and follow up on issues in the zairachem-docker repo
